datareservoirio/_logging.py

```diff
@@ -41,3 +41,25 @@ def wrapper(self, *args, **kwargs):
             raise e
 
     return wrapper
+
+
+def log_retry(func):
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        try:
+            return func(self, *args, **kwargs)
+        except Exception as e:
```

Comment:
If we only want to log a certain type of exceptions, we need to add that to the filter here, that is capture only the specific Exceptions we want to log with giving it all the ones we want to catch https://docs.python.org/3/tutorial/errors.html#handling-exceptions